tests/e2e/tests/base/run-different-analysis.test.ts

```diff
@@ -0,0 +1,71 @@
+import * as pathlib from 'path';
+import { RepoData, expect, test } from '../../fixtures/test-repo-fixture';
+import { OPENAI_GPT4O_PROVIDER } from '../../fixtures/provider-configs.fixture';
+import { readFileSync } from 'fs';
+import * as VSCodeFactory from '../../utilities/vscode.factory';
+
+// Load test repos data
+const testReposPath = pathlib.join(__dirname, '../../fixtures/test-repos.json');
+const testReposData: RepoData = JSON.parse(readFileSync(testReposPath, 'utf-8'));
+
+test.describe('Run analysis for different repositories', () => {
+  const entries = Object.entries(testReposData) as [keyof RepoData, RepoData[keyof RepoData]][];
+
+  for (const [repoKey, repoInfo] of entries) {
+    if (repoKey === 'jboss-eap-quickstarts') {
+      continue;
+    }
+    test(`Analyze ${String(repoKey)} app`, async ({}, testInfo) => {
+      test.setTimeout(900000);
+      const profileName = `${String(repoKey)} analysis`;
+
+      const vscodeApp = await VSCodeFactory.open(
+        repoInfo.repoUrl,
+        repoInfo.repoName,
+        repoInfo.branch
+      );
+
+      try {
+        await test.step('Create profile', async () => {
+          await vscodeApp.createProfile(repoInfo.sources, repoInfo.targets, profileName);
+        });
+
+        await test.step('Start KAI server', async () => {
+          await vscodeApp.startServer();
+        });
+
+        await test.step('Run analysis', async () => {
+          await vscodeApp.runAnalysis();
+          await vscodeApp.waitForAnalysisCompleted();
+        });
+
+        await test.step('Verify issues and incidents counts', async () => {
+          if (repoInfo.issuesCount === undefined) {
+            throw new Error(
+              `'issuesCount' should be set for ${repoInfo.repoName} in test-repos.json`
+            );
+          }
+          if (repoInfo.incidentsCount === undefined) {
+            throw new Error(
+              `'incidentsCount' should be set for ${repoInfo.repoName} in test-repos.json`
+            );
+          }
```

Comment:
Can we move this to the beginning of the tests?

It doesn't make sense to perform all the configuration and run the analysis if we'll fail right after the analysis because the developer didn't specify the expected count